sklearn/utils/_mask.py

```diff
@@ -1,19 +1,41 @@
 import numpy as np
+from scipy import sparse
 
 from . import is_scalar_nan
 from .fixes import _object_dtype_isnan
 
 
-def _get_mask(X, value_to_mask):
-    """Compute the boolean mask X == value_to_mask."""
+def _fit_mask(X, value_to_mask):
     if is_scalar_nan(value_to_mask):
         if X.dtype.kind == "f":
-            return np.isnan(X)
+            Xt = np.isnan(X)
         elif X.dtype.kind in ("i", "u"):
             # can't have NaNs in integer array.
-            return np.zeros(X.shape, dtype=bool)
+            Xt = np.zeros(X.shape, dtype=bool)
         else:
             # np.isnan does not work on object dtypes.
-            return _object_dtype_isnan(X)
+            Xt = _object_dtype_isnan(X)
     else:
-        return X == value_to_mask
+        Xt == value_to_mask
+    return Xt
+
+
+def _get_mask(X, value_to_mask, reconstruct_sparse=False):
+    """Compute the boolean mask X == missing_values."""
+    # We get entire sparse matrix when reconstruct is True
```

Comment:
Instead to have a comment here, it could be better to have it inside the above docstring of the function

```
Parameters
----------
X :....
reconstruct_sparse : bool, default=False
    Whether or not we need to reconstruct the sparse matrix. If True ....
```

Reply:
Done